vendor/bat-native-ads/src/bat/ads/internal/ad_events/ad_event_util.h

```diff
@@ -8,12 +8,53 @@
 
 #include "bat/ads/internal/ad_events/ad_event_info.h"
 
+#include <algorithm>
+
+#include "base/time/time.h"
+
 namespace ads {
 
 struct AdInfo;
 
 bool HasFiredAdViewedEvent(const AdInfo& ad, const AdEventList& ad_events);
 
+template <typename T>
+base::Time GetLastSeenAdTime(const AdEventList& ad_events,
+                             const T& creative_ad) {
```

Comment:
Could we pass `const std::string& creative_instance_id` here instead `creative_ad`?
In this case we can do this function as a not template.

Reply:
@aseren we've had it as such, but @tmancey argued for passing the `creative_ad` for better extensibility. @tmancey wdyt?

Reply:
We should change to `const CreativeAdInfo& creative_ad`